jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java

```diff
@@ -1368,7 +1368,9 @@ public static boolean isValidHostRegisteredName(String token)
      * @param path the URI path
      * @param query the URI query
      * @return A String URI
+     * @deprecated use {@code HttpScheme} from {@code jetty-http} instead (as it does a proper job with normalization).  Will be removed in Jetty 12.1.0.
```

Comment:
Remove `Will be removed in Jetty 12.1.0` as we don't even know if we're going to have that release (may skip to 13 due to minimum Java requirement at 21).